ls1intum/Artemis
# `Team exercises`: Add csv file team import and provide better error messages

### Checklist
#### General


#### Client

### Motivation and Context


Closes #4255 
Closes #4256 
In order to use external tools to create teams of students for an exercise, Artemis has an import teams function. This function however only supports json files. Tools like Moodle however export their teams as csv files. So it would be useful to be able to import these files directly.
Also, the error messages when the imported file was malformed were quite poor. Since these files can contains hundreds or thousands of entries it is important to at least tell the user which entry is responsible for the error.

### Description

For the import of csv files I simply changed the file import to also accept csv files which are then parsed with PapaParse. They accept the same keys as the json file requires, but also accept some additional keys like `Matrikelnummer` instead of `registrationNumber`. For example this will allow instructors to directly import files from Moodle (The format of which was provided in #1994)
For the error messages I changed the language keys to also include parameters that describe which entry is responsible for the error. The error messages now look similar to this:
> The team name 1Team XYZ in entry 101 does not start with a letter.

### Steps for Testing

Prerequisites:
- 1 Instructor
- A course with a couple of registered students
- 1 Exercise with teams enabled
- Some valid and invalid json files (You can create these by exporting teams from another exercise)
- Some valid and invalid csv files (You can create these by exporting teams from another exercise and convert the json file here: https://www.convertcsv.com/json-to-csv.htm)

1. Log in to Artemis
2. Navigate to the course management exercise page of the exercise that has teams enabled
3. Navigate to the teams page of this exercise
4. Click on Import Teams
5. Import the csv or json files
    - The valid files should import with no problem
    - The invalid files should provide proper error messages

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage





File: Lines % | Branches %
teams-import-from-file-form.component.ts:  95.95% | 73.33%

### Screenshots


No direct UI changes. Only the error messages are different.
